src/Microsoft.ML.PipelineInference/PipelinePattern.cs

```diff
@@ -183,57 +196,107 @@ public Experiment CreateTrainTestExperiment(IDataView trainData, IDataView testD
         /// <summary>
         /// Runs a train-test experiment on the current pipeline, through entrypoints.
         /// </summary>
-        public double RunTrainTestExperiment(IDataView trainData, IDataView testData, AutoInference.SupportedMetric metric, MacroUtils.TrainerKinds trainerKind)
+        public void RunTrainTestExperiment(IDataView trainData, IDataView testData,
+            AutoInference.SupportedMetric metric, MacroUtils.TrainerKinds trainerKind, out double testMetricValue,
+            out double trainMetricValue)
         {
-            var experiment = CreateTrainTestExperiment(trainData, testData, trainerKind, out var trainTestOutput);
+            var experiment = CreateTrainTestExperiment(trainData, testData, trainerKind, true, out var trainTestOutput);
             experiment.Run();
+
             var dataOut = experiment.GetOutput(trainTestOutput.OverallMetrics);
             var schema = dataOut.Schema;
-            schema.TryGetColumnIndex(metric.Name, out var metricCol);
+            _env.Check(schema.TryGetColumnIndex(metric.Name, out var metricCol),
+                $"Schema does not contain metric column: {metricCol}");
+            double metricValue = 0;
+            double trainingMetricValue = 0;
 
             using (var cursor = dataOut.GetRowCursor(col => col == metricCol))
             {
                 var getter = cursor.GetGetter<double>(metricCol);
-                double metricValue = 0;
-                cursor.MoveNext();
+                bool moved = cursor.MoveNext();
+                _env.Check(moved, "Could not move cursor forward. (No rows to extract.)");
                 getter(ref metricValue);
-                return metricValue;
+                _env.Check(!cursor.MoveNext(), "Encountered additional, unexpected row.");
+            }
```

Comment:
you use this code pattern twice here, and once in AutoMlUtils, maybe it make sense to refactor it to a method?

Reply:
Good suggestion. I'll make that change.